### ISSUE TITLE
Incorrect nullability warnings for tuples

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#61918]
**Version Used**: 

```
Microsoft Visual Studio Professional 2022
Version 17.3.0 Preview 1.1
VisualStudio.17.Preview/17.3.0-pre.1.1+32519.111
Microsoft .NET Framework
Version 4.8.04084
```

**Steps to Reproduce**:

Compile the following code
```cs

#nullable enable
class C
{
  void Foo((int[]? A, int B) x)
  {
    if (x.A == null) return;

    Bar(x); // CS8620: Argument of type '(int[]? A, int B)' cannot be used for parameter 'tuple' of type '(int[] A, int B)' in 'void C.Bar((int[] A, int B) tuple)' due to differences in the nullability of reference types.
    Bar((x.A, x.B)); // no warnings
  }

  private void Bar((int[] A, int B) tuple) { }
}
``` 

**Expected Behavior**:
No warnings. The tuple component `x.A` is known to be not nullable.
`Bar(x)` and `Bar((x.A, x.B))` should be equivalent since the latter simply reconstructs an equivalent tuple from the same component values.

**Actual Behavior**:
Warning for `Bar(x)`: 
```
// CS8620: Argument of type '(int[]? A, int B)' cannot be used for parameter 'tuple' of type '(int[] A, int B)' in 'void C.Bar((int[] A, int B) tuple)' due to differences in the nullability of reference types.
```

**Comment by RikkiGibson** [comment]
This warning might be occurring because the struct type is invariant. We could consider making tuples special--e.g. `(int[]?, int)` can convert to `(int[], int)` if the flow states are compatible. We could also consider doing a special analysis of some sort of the underlying fields for all struct types. The latter could be expensive.
